pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -371,45 +370,44 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		allNodeNames = append(allNodeNames, name)
 	}
 
-	var lock sync.Mutex
-	var firstError error
-
-	topologyMaps := newTopologyPairsMaps()
-
-	appendTopologyPairsMaps := func(toAppend *topologyPairsMaps) {
-		lock.Lock()
-		defer lock.Unlock()
-		topologyMaps.appendMaps(toAppend)
-	}
-	catchError := func(err error) {
-		lock.Lock()
-		defer lock.Unlock()
-		if firstError == nil {
-			firstError = err
-		}
-	}
+	allNodesCount := len(allNodeNames)
+	errCh := make(chan error, 1)
+	topologyPairsCh := make(chan *topologyPairsMaps, allNodesCount)
 
 	ctx, cancel := context.WithCancel(context.Background())
 
 	processNode := func(i int) {
 		nodeInfo := nodeInfoMap[allNodeNames[i]]
 		node := nodeInfo.Node()
 		if node == nil {
-			catchError(fmt.Errorf("node not found"))
+			catchError(errCh, fmt.Errorf("node not found"))
```

Comment:
let's output the node name as well:

```suggestion
			catchError(errCh, fmt.Errorf("node %q not found", node.Name))
```

Reply:
and you can also short circuit here by `cancel()`.

Reply:
https://github.com/kubernetes/kubernetes/pull/79774 do the short circuit, and I'll rebase after it gets merged.